internal/telemetry/event.go

```diff
@@ -0,0 +1,66 @@
+package telemetry
+
+import (
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Event is a telemetry event
+type Event struct {
+	id          string
+	eventType   EventType
+	userID      string
+	time        time.Time
+	executionID string
+	command     string
+	data        map[DataKey]interface{}
+}
+
+// NewCommandStartEvent creates a new event for a command's start
+func NewCommandStartEvent(command string) Event {
+	data := make(map[DataKey]interface{})
+	return NewEvent(EventTypeCommandStart, command, data)
+}
+
+// NewCommandCompleteEvent creates a new event for a command's start
+func NewCommandCompleteEvent(command string) Event {
+	data := make(map[DataKey]interface{})
+	return NewEvent(EventTypeCommandStart, command, data)
+}
+
+// NewCommandErrorEvent creates a new event for a command that has errored
+func NewCommandErrorEvent(command string, err error) Event {
+	data := make(map[DataKey]interface{})
+	data[DataKeyErr] = err
+	return NewEvent(EventTypeCommandStart, command, data)
+}
+
+// NewEvent creates a new event
+func NewEvent(eventType EventType, command string, data map[DataKey]interface{}) Event {
```

Comment:
[nit/question] let's move `command string` ahead of `eventType EventType` in the arg positioning.  although i thought `command` was something we'd be able to construct our tracker with, along with `userID`?

with my previous comment, looking for something like this:

```go
func NewEvent(command string, eventType EventType, data ...EventData) Event {
}

type EventData struct {
  Key EventDataKey // renamed for consistency with the rest
  Value interface{}
}
```

Reply:
do you want me to move configureTelemetry out of the setup function and into the build function then so we can do that?

Reply:
yeah, that sounds fine by me, lets put it right b/w `display` initialization and the `cmd` declaration